src/Member/MemberUtil.php

```diff
@@ -172,4 +179,43 @@ public function findOrCreate(string $email)
 
         return $member;
     }
+
+    /**
+     * @return \Contao\Model[]|null
+     */
+    public function getActiveGroups(int $memberId, array $options = [])
+    {
+        if (!$memberId) {
+            return null;
+        }
+
+        if (null === ($memberModel = $this->modelUtil->findModelInstanceByIdOrAlias('tl_member', $memberId, $options))) {
+            return null;
+        }
+
+        if (null === ($groups = StringUtil::deserialize($memberModel->groups, true))) {
+            return null;
+        }
+
+        $columns = ['tl_member_group.id IN('.implode(',', array_map('\intval', $groups)).')'];
+
+        $this->modelUtil->addPublishedCheckToModelArrays('tl_member_group', 'disable', 'start', 'stop', $columns, ['invertPublishedField' => true]);
+
+        if (null === ($groupModelCollection = $this->modelUtil->findModelInstancesBy('tl_member_group', $columns, []))) {
+            return null;
+        }
+
+        return $groupModelCollection->getModels();
+    }
+
+    public function hasActiveGroups(int $memberId): bool
```

Comment:
This was not the intention (also not yours I guess).

The intention was:

`public function hasActiveGroups(int $memberId, array $groups = []): bool`

But in fact here we can't say what to return if only some of the groups are available. Hence please refactor it to

`public function hasActiveGroup(int $memberId, int $group): bool`

and check if the group is in the array of active groups.

Thanks (for UserUtil also).